src/redis-cli.c

```diff
@@ -1532,6 +1536,24 @@ static int parseOptions(int argc, char **argv) {
             i = j;
         } else if (!strcmp(argv[i],"--cluster") && lastarg) {
             usage();
+        } else if ((!strcmp(argv[i],"--cluster-only-masters") && !lastarg)) {
```

Comment:
The `--cluster-only-masters` and `--cluster-only-slaves` options should only activate a flag, they don't need any argument. So the check for `!lastarg` should be removed. Also, the command (in this case being `call`) should not follow the `--cluster-only-*` options.

The syntax of the CL command here should be something like this: 

`redis-cli --cluster call REDIS_COMMAND --cluster-only-masters`

or

`redis-cli --cluster call --cluster-only-masters REDIS_COMMAND`

This is the standard syntax for --cluster mode.

Reply:
resolved